promotion/datastore.go

```diff
@@ -908,6 +939,91 @@ ORDER BY created_at DESC
 	return nil, nil
 }
 
+// RunNextBatchPaymentsJob to sign claim credentials if there is a claim waiting, returning true if a job was attempted
+func (pg *Postgres) RunNextBatchPaymentsJob(ctx context.Context, worker BatchTransferWorker) (bool, error) {
+
+	// setup a logger
+	logger, err := appctx.GetLogger(ctx)
+	if err != nil {
+		// no logger, setup
+		ctx, logger = logging.SetupLogger(ctx)
+	}
+
+	// create a tx
+	tx, err := pg.RawDB().Beginx()
+	attempted := false
+	if err != nil {
+		return attempted, err
+	}
+	defer pg.RollbackTx(tx)
+
+	// until nitro payments integration, this is just for bitflyer
+	statement := `
+		select
+			cd.batch_id
+		from
+			claim_drain cd
+			join wallets w on w.id=cd.wallet_id
+		where
+			cd.erred = false and
+			cd.status='prepared' and
+			w.user_deposit_account_provider = 'bitflyer'
+		group by
+			cd.batch_id
+		having
+			bool_and(cd.transaction_id is not null)=true
+		for update skip locked
+		limit 1;
+`
+	var batchID *uuid.UUID
+
+	err = tx.Select(&batchID, statement)
+	if err != nil {
+		return attempted, err
+	}
+
+	if batchID == nil {
+		return attempted, nil
+	}
+
+	attempted = true
+
+	// perform submit against payments API
+	err = worker.SubmitBatchTransfer(ctx, batchID)
+	if err != nil {
+		// log the error from redeem and transfer
+		logger.Error().Err(err).Msg("failed to redeem and transfer funds")
+		status, errCode, _ := errToDrainCode(err)
+		_, err = tx.Exec(`
+			update claim_drain set
+				erred = true,
+				errcode = $1,
+				status = $2
+			where batch_id = $3`, errCode, status, batchID)
+		if err != nil {
+			return attempted, err
+		}
+		// inform sentry about this error
+		sentry.CaptureException(err)
```

Comment:
should we enrich the error with some more info like batch id and maybe the job name. Might apply to some of the other errors as well

Reply:
https://github.com/brave-intl/bat-go/pull/973/commits/804b092ef331a7fa090c0dd7b48b391e2fb5a1fc addressed here, think the error classification is good for sentry, dont think we want in depth details.